xrdp/xrdp.c

```diff
@@ -149,8 +148,8 @@ xrdp_shutdown(int sig)
     tbus threadid;
 
     threadid = tc_get_threadid();
-    g_writeln("shutting down");
-    g_writeln("signal %d threadid %lld", sig, (long long)threadid);
+    LOG(LOG_LEVEL_INFO, "shutting down");
```

Comment:
Since the work being done in this function is signal safe except for the logging, I think it makes sense to just fix the problem. Looking a bit more at the code it make sense for the logging to be where the term signal is checked and termination business logic is executed.

Reply:
Agreed.